basin3d/plugins/usgs.py

```diff
@@ -420,36 +404,39 @@ def get_hydrological_unit_codes(self):
 
         return usgs_huc_codes.CONTENT
 
-    def get(self, pk=None, **kwargs):
+    def get(self, query: QueryById):
         """ Get a single Region object
 
         ===================== === =====================
         USGS NWIS                 Broker
         ===================== === =====================
-        ``{huc}``              >>  ``Primary key (pk)``
+        ``{huc}``              >>  ``Primary key (query.id)``
         --------------------- --- ---------------------
-        ``new_huc_rdb.txt``    >>  ``monitoringfeatures/<feature_type>/{pk}``
+        ``new_huc_rdb.txt``    >>  ``monitoringfeatures/<feature_type>/{query.id}``
         ===================== === =====================
 
-        :param request: The request object ( Please refer to the Django documentation )
-        :param pk: USGS Daily Values Plugin primary key
+        :param query: The query info object
         :return: a serialized ``MonitoringFeature`` object
         """
-        feature_type = None
-        if QUERY_PARAM_FEATURE_TYPE in kwargs and kwargs[QUERY_PARAM_FEATURE_TYPE]:
-            feature_type = kwargs[QUERY_PARAM_FEATURE_TYPE]
+        if len(query.id) == 2:
+            mf_query = QueryMonitoringFeature(monitoring_features=[query.id], feature_type=FeatureTypeEnum.REGION)
+        elif len(query.id) == 4:
+            mf_query = QueryMonitoringFeature(monitoring_features=[query.id], feature_type=FeatureTypeEnum.SUBREGION)
+        elif len(query.id) == 6:
+            mf_query = QueryMonitoringFeature(monitoring_features=[query.id], feature_type=FeatureTypeEnum.BASIN)
+        elif len(query.id) == 8:
+            mf_query = QueryMonitoringFeature(monitoring_features=[query.id], feature_type=FeatureTypeEnum.SUBBASIN)
         else:
-            if len(pk) == 2:
-                feature_type = FeatureTypes.TYPES[FeatureTypes.REGION]
-            elif len(pk) == 4:
-                feature_type = FeatureTypes.TYPES[FeatureTypes.SUBREGION]
-            elif len(pk) == 6:
-                feature_type = FeatureTypes.TYPES[FeatureTypes.BASIN]
-            elif len(pk) == 8:
-                feature_type = FeatureTypes.TYPES[FeatureTypes.SUBBASIN]
-
-        for o in self.list(pk=pk, feature_type=feature_type):
+            mf_query = QueryMonitoringFeature(monitoring_features=[query.id], feature_type=FeatureTypeEnum.POINT)
+
+        for o in self.list(query=mf_query):
             return o
+
+        # An 8 character code can also be a point, Try that
+        if len(query.id) == 8:
+            for o in self.list(query=QueryMonitoringFeature(monitoring_features=[query.id],
+                                                            feature_type=FeatureTypeEnum.POINT)):
+                return o
```

Comment:
OK for now. I'd like to do a bit more testing. I thought we had an issue for this but am not seeing it. I'd like to make one and reference it in this PR or commit history (if not already).

Reply:
When we had our review, I had forgotten that I had already done this.  I don't remember if an issue was created...